src/librustc_driver/lib.rs

```diff
@@ -1431,6 +1444,58 @@ pub fn in_rustc_thread<F, R>(f: F) -> Result<R, Box<Any + Send>>
     thread.unwrap().join()
 }
 
+/// Get a list of extra command-line flags provided by the user, as strings.
+///
+/// This function is used during ICEs to show more information useful for
+/// debugging, since some ICEs only happens with non-default compiler flags
+/// (and the users don't always report them).
+fn extra_compiler_flags() -> Option<Vec<String>> {
+    let mut args = Vec::new();
+    for arg in env::args_os() {
+        if let Ok(string) = arg.into_string() {
+            args.push(string);
+        } else {
+            return None;
```

Comment:
Hm, I feel like we shouldn't just return None but maybe just use `into_string_lossy()` here? That seems more reasonable...

Reply:
Uh, yeah, fixed.